### ISSUE TITLE
Since NS7 update, some plugins fail building with "Error: spawn ./gradlew EACCES"

### ISSUE THREAD

**Issue by SzaboKrisztian** [NativeScript/nativescript-cli#5368]
I personally haven't tested it on Windows, so there might not be any problems on that platform because the build scripts are handled separately.

Since the update to NS7 a few days ago, there have been multiple people on the NSCommunity Slack reporting this. I personally have encountered the problem in relation to `nativescript-socket.io`, but it's not specific to it, as it actually boils down to an issue in the CLI.

The problem is two-fold: first, the EACCES error is caused by the `gradlew` file not being being marked as executable. Once this is fixed, errors relating to the carriage return character start popping up. The issue now is that the `gradlew` file uses windows style newline (`\r\n`) instead of *nix style newline (`\n`). I've implemented a quick and dirty fix, but this only solves the symptoms, it doesn't solve the problem at the root cause as I'm not familiar enough with NS to know which part of the tooling generates the `gradlew` file when building plugins.

The temporary fix involves editing a file in the NS CLI: 
1. in the global node_modules folder (unless NS is locally installed under the project's devDependencies), find `nativescript/lib/services/android-plugin-build-service.js`
2. anywhere at the top of the file (maybe between lines 19-24 for consistency's sake) add `const fs = require("fs");`
3. at line 413 (formerly 412 before the above addition) add:
```javascript
if (!this.$hostInfo.isWindows) {
    const gradleFile = path.join(pluginBuildSettings.pluginDir, gradlew);
    let content = fs.readFileSync(gradleFile, {encoding:'utf8', flag:'r'});
    content = content.replace(/\r/g, '');
    fs.writeFileSync(gradleFile, content);
    fs.chmodSync(gradleFile, 0o755, (err) => {
        if (err) {
            console.log("Error changing setting gradlew as executable");
            throw err;
        }
    });
}
```

**Comment by rigor789** [comment]
I believe https://github.com/NativeScript/nativescript-cli/releases/tag/v7.0.6 should no longer have the issue. 

I verified the package before publishing and it has `LF` line endings.

**Comment by romandragan** [comment]
@rigor789, any progress on this?

**Comment by rigor789** [comment]
Yep, fixes have been merged to master, and will release a regular version very soon - until then, feel free to use the `next` tag, or the full version mentioned above.

**Comment by rigor789** [comment]
I'm not sure the latest pr #5382 is really necessary after adjusting the repo's `.gitattributes` the line endings and file attributes should remain intact regardless of the OS the repo is checked out from. But I guess it doesn't hurt as an extra precaution.
